source/coding-the-legislation/legislation_parameters.md

```diff
@@ -95,43 +95,55 @@ public_sector:
      ...
 ```
 
-#### Create Scales
+#### Scales
 
-Scales are constituted of brackets. Brackets are defined by amounts, bases, rates, average rates and thresholds.
+Scales are complex parameters constituted of brackets. They offer convenient built-in tools for recurring calculation patterns.
 
-Sample `parameters/tax_on_salary/tax_scale.yaml`:
+For instance, a marginal rate test scale can be defined in a YAML file with:
+
+File `parameters/tax_on_salary/tax_scale.yaml`
 ```yaml
 description: Scale for tax on salaries
 brackets:
-- rate:
-    1950-01-01:
-      value: 0.0
-    2010-01-01:
-      value: 0.02
-  threshold:
-    1950-01-01:
-      value: 0.0
-- rate:
-    1950-01-01:
-      value: 0.2
-  threshold:
-    1950-01-01:
-      value: 2000
+  - rate: # Define a 1st bracket
+      1950-01-01:
+        value: 0.0  # The rate applied to the first bracket was 0% from 1950 to 2009
+      2010-01-01:
+        value: 0.02  # The rate applied to the first bracket has changed to 2% in 2010
+    threshold:
+      1950-01-01:
+        value: 0.0
+  - rate: # Define a 2nd bracket
+      1950-01-01:
+        value: 0.2  # The rate applied to the second bracket has been 20% since 1950
+    threshold:
+      1950-01-01:
+        value: 2000  # The 2nd bracket starts for wages beyond 2000
+metadata:
+  threshold_unit: currency-EUR
+  rate_unit: /1
 ```
 
-Example: [the french tax scale on income](https://fr.openfisca.org/legislation/impot_revenu.bareme)
-
-#### Import parameters from IPP tables
-
-> This section applies only to OpenFisca-France.
+It can then be used in a formula with:
+```py
+def formula(person, period, parameters):
+    salary = person('salary', period)
+    scale = parameters(period).tax_on_salary.tax_scale
+    return scale.calc(salary)
+```
 
-The [<abbr title="Institut des politiques publiques">IPP</abbr>](http://www.ipp.eu/) is a French centre in economics which produces tax and benefit tables in the `XLSX` format, with parameters history.
+Other built-in scales are:
```

Comment:
I'd prefer if this was "Here are all the possible scales…" even at the cost of some repetition of the information on MarginalRateScale.

Reply:
👍